src/Tracing/DynamicSamplingContext.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Tracing;
+
+use Sentry\State\HubInterface;
+use Sentry\State\Scope;
+
+/**
+ * This class represents the Dynamic Sampling Context (dsc).
+ *
+ * @see https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/
+ */
+final class DynamicSamplingContext
+{
+    public const SENTRY_ENTRY_PREFIX = 'sentry-';
+
+    /**
+     * The dsc entries.
+     *
+     * @var array<string, string>
+     */
+    private $entries;
+
+    /**
+     * Indicates that the dsc is frozen and cannot be mutated.
+     *
+     * @var bool
+     */
+    private $frozen = false;
+
+    /**
+     * Construct a new dsc object.
+     */
+    private function __construct()
+    {
+        $this->entries = [];
+    }
+
+    /**
+     * Set a new key value pair on the dsc.
+     *
+     * @param string $key   the list member key
+     * @param string $value the list member value
+     */
+    public function set(string $key, string $value): void
+    {
+        if ($this->frozen) {
+            return;
+        }
+
+        $this->entries[$key] = $value;
+    }
+
+    /**
+     * Check if a key value pair is set on the dsc.
+     *
+     * @param string $key the list member key
+     */
+    public function has(string $key): bool
+    {
+        return isset($this->entries[$key]);
+    }
+
+    /**
+     * Get a value from the dsc.
+     *
+     * @param string      $key     the list member key
+     * @param string|null $default the default value to return if no value exists
+     */
+    public function get(string $key, ?string $default = null): ?string
+    {
+        return $this->entries[$key] ?? $default;
+    }
+
+    /**
+     * Mark the dsc as frozen.
+     */
+    public function freeze(): void
+    {
+        $this->frozen = true;
+    }
+
+    /**
+     * Indicates that the dsc is frozen and cannot be mutated.
```

Comment:
nit(again): This comment is not really describing the variable.

It can be something like:
Identigies weather the dynamic sampling context is mutable or not